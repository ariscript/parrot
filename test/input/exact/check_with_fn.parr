def factorial(n, acc):
    if n == 0: acc
    else: factorial(n - 1, acc * n)

check:
    val fact = (lambda (n): factorial(n, 1)),
    fact(1) is 1,
    fact(3) is 6 because 3 * 2 * 1,
    fact(4) is 24 because 4 * fact(3),
    fact(5) is -3 because ().

nil
