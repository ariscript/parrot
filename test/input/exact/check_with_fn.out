failed test at 10:4-10:28: the explanation and right side were not `equal` (() and -3)
failed test at 10:4-10:28: the left and right sides were not `equal` (120 and -3)
check block: 3/4 tests passed
nil
